extensions/transferdetails/service.go

```diff
@@ -0,0 +1,272 @@
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := deriveDIDs(req.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	// overwriting is not enough because it is not required that
+	// the funding payload contains all funding attributes
+	model, err = extensions.DeleteAttributesSet(model, TransferDetailData{}, idx, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+
+	attributes, err := extensions.FillAttributeList(*req.Data, idx, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+
+	err = model.AddAttributes(
+		documents.CollaboratorsAccess{
+			ReadWriteCollaborators: c,
+		},
+		true,
+		attributes...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	validator := CreateValidator()
+	err = validator.Validate(nil, model)
+	if err != nil {
+		return nil, errors.NewTypedError(documents.ErrDocumentInvalid, err)
+	}
+
+	return model, nil
+}
+
+// TODO: generic?
+func (s service) findTransfer(model documents.Model, transferID string) (*TransferDetailData, error) {
+	idx, err := extensions.FindAttributeSetIDX(model, transferID, transfersLabel, transferIDLabel, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+	return s.deriveTransferData(model, idx)
+}
+
+// TODO: generic?
+func (s service) deriveTransferData(model documents.Model, idx string) (*TransferDetailData, error) {
+	data := new(TransferDetailData)
+
+	types := reflect.TypeOf(*data)
+	for i := 0; i < types.NumField(); i++ {
+		// generate attr key
+		jsonKey := types.Field(i).Tag.Get("json")
+		label := extensions.LabelFromJSONTag(idx, jsonKey, transfersFieldKey)
+
+		attrKey, err := documents.AttrKeyFromLabel(label)
+		if err != nil {
+			return nil, err
+		}
+
+		if model.AttributeExists(attrKey) {
+			attr, err := model.GetAttribute(attrKey)
+			if err != nil {
+				return nil, err
+			}
+
+			// set field in data
+			n := types.Field(i).Name
+
+			v, err := attr.Value.String()
+			if err != nil {
+				return nil, err
+			}
+
+			reflect.ValueOf(data).Elem().FieldByName(n).SetString(v)
+
+		}
+
+	}
+	return data, nil
+}
+
+// DeriveTransferResponse returns create response from the added transfer detail
+func (s service) DeriveTransferResponse(ctx context.Context, model documents.Model, transferID string) (*TransferDetailResponse, error) {
+	idx, err := extensions.FindAttributeSetIDX(model, transferID, transfersLabel, transferIDLabel, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+
+	h, err := documents.DeriveResponseHeader(s.tokenRegistry, model)
+	if err != nil {
+		return nil, errors.New("failed to derive response: %v", err)
+	}
+	data, err := s.deriveTransferData(model, idx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &TransferDetailResponse{
+		Header: h,
+		Data:   data,
+	}, nil
+
+}
+
+// DeriveTransfersListResponse returns a transfers list
+func (s service) DeriveTransferListResponse(ctx context.Context, model documents.Model) (*TransferDetailListResponse, error) {
+	response := new(TransferDetailListResponse)
```

Comment:
Same here:

We only need an additional parameter for the struct to be generic. like ` TransferDetailData` or `Funding`